nri-prelude/src/Task.hs

```diff
@@ -198,6 +200,23 @@ parallel tasks =
           |> Shortcut.map Prelude.sequence
     )
 
+-- | Given a task and a callback, execute the task in a greenthread
+-- and sends its result to callback.
+-- Always returns @OK ()@
+background :: forall x a . Task x a -> (Result x a -> Task x a) -> Task x ()
```

Comment:
would an additional variant that allows us to return a (wrapped) async value that we can then await on be useful?  feel free to veto if it's too low-level for our library

Reply:
I think that'd be really cool, but how would usage be? I like the current implementation because it keeps us within the Task monad with no user overhead/transformation/handlers.

Reply:
something like

```haskell
task :: Task x a
task = <...>

main = do
  asyncedTask <- ourAsync task
  result <- ourAwait asyncedTask
  doSomethingWith result
```

basically a `Task`-flavoured `async` / `await` that allows us to stay within `Task`.  this can be a bit error prone to use, though, for the reasons we should be using `withAsync` when dealing with `Async`s.  i do think we should _keep_ `background` as it is (modulo any other requested fixes); the `ourAsync` / `ourAwait` (or whatever) would exist beside `background`.  if it's not worth the effort, don't worry bout it

Reply:
Very much like it, but would prefer to do it in another PR